lib/srv/app/session.go

```diff
@@ -94,15 +121,33 @@ func (s *Server) newSession(ctx context.Context, identity *tlsca.Identity, app t
 		return nil, trace.Wrap(err)
 	}
 
-	return &session{
-		fwd:          fwd,
-		streamWriter: streamWriter,
-	}, nil
+	// Put the session chunk in the cache so that upcoming requests can use it for
+	// 5 minutes or the time until the certificate expires, whichever comes first.
+	ttl := utils.MinTTL(identity.Expires.Sub(s.c.Clock.Now()), 30*time.Second)
```

Comment:
Nit: The comment says 5 minutes but here TTL is set to 30 seconds which is a little confusing.